tools/clang/lib/SPIRV/SPIRVEmitter.cpp

```diff
@@ -1645,6 +1663,35 @@ uint32_t SPIRVEmitter::processBinaryOp(const Expr *lhs, const Expr *rhs,
   return 0;
 }
 
+void SPIRVEmitter::initOnce(std::string varName, uint32_t varPtr,
+                            const Expr *varInit) {
+  const uint32_t boolType = theBuilder.getBoolType();
+  varName = "init.done." + varName;
+
+  // Create a file/module visible variable to hold the initialization state.
+  const uint32_t initDoneVar = theBuilder.addFileVar(
```

Comment:
I think each invocation of the shader has its own copy of `Private` stuff? But I'm 100% sure. Need to confirm with others.

Glslang just ... ignores the `static` keyword AFAICT.